src/OrchardCore.Modules/OrchardCore.Workflows/Controllers/WorkflowTypeController.cs

```diff
@@ -124,11 +129,20 @@ public async Task<IActionResult> Index(WorkflowTypeIndexOptions options, PagerPa
                 .Take(pager.PageSize)
                 .ListAsync();
 
-            var workflowTypeIds = workflowTypes.Select(x => x.WorkflowTypeId).ToList();
-            var workflowGroups = (await _session.QueryIndex<WorkflowIndex>(x => x.WorkflowTypeId.IsIn(workflowTypeIds))
-                .ListAsync())
-                .GroupBy(x => x.WorkflowTypeId)
-                .ToDictionary(x => x.Key);
+            var workflowIndexTable = $"{_session.Store.Configuration.TablePrefix}{nameof(WorkflowIndex)}";
+            var dialect = _session.Store.Configuration.SqlDialect;
+            var selectSql = $"select distinct {dialect.QuoteForColumnName(nameof(WorkflowIndex.WorkflowTypeId))} from {dialect.QuoteForTableName(workflowIndexTable, _session.Store.Configuration.Schema)}";
```

Comment:
If this needs to be an SQL string, then uppercase the SQL keywords and split it into multiple lines for readability (now it's really long).